### ISSUE TITLE
Add support for range formatting

### ISSUE THREAD

**Issue by jvanbruegge** [dhall-lang/dhall-haskell#1588]
Due to my issues with the current pretty printer (#1099, but also #1496), I would like to format only a subset of lines. The language server protocol has `textDocument/rangeFormatting` for this, but it is currently not implemented by the dhall lsp

**Comment by Gabriel439** [comment]
@jvanbruegge: Yeah, the Dhall API does not support formatting a subset of lines unless that subset of lines represented a well-formed expression in isolation

**Comment by jvanbruegge** [comment]
Yeah, but would it be possible to find the next complete expression from a range? When I use range formatting, I usually just want format one let (but a complete let)

**Comment by Gabriel439** [comment]
@jvanbruegge: Yeah, I believe it should be possible because we store range annotations in our parsed syntax tree which we could use to infer the desired complete range.  However, I might not have time to implement this myself
